### ISSUE TITLE
Wrong predictions after training structured data classification using keras

### ISSUE THREAD

**Issue by sandeepreddy5** [keras-team/keras-io#1341]
Hi,

I've followed [**Structured data classification from scratch**](https://keras.io/examples/structured_data/structured_data_classification_from_scratch/) but custom dataset and trained a model with target variable as 0 and 1. but it's giving output greater than 1 like 4,5 even up to 8.
How can i fix it to get only 0 and 1 as target values.

Thank you

**Comment by SuryanarayanaY** [comment]
Hi @sandeepreddy5 ,

There might be two possibilities. Could you please check the no of labels in your dataset once again. Providing sample dataset may help us to verify it.If you are using a dataframe df and label as Target then df['Target'].unique() may output number of target variables in the dataframe.

Could you please check the activation function you are using on the output layer.It should be sigmoid like below.

`output = layers.Dense(1, activation="sigmoid")(x)`
